core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXV1EventController.java

```diff
@@ -0,0 +1,75 @@
+package com.sequenceiq.distrox.v1.distrox.controller;
+
+import static com.sequenceiq.cloudbreak.common.exception.NotFoundException.notFound;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.authorization.annotation.CheckPermissionByResourceCrn;
+import com.sequenceiq.authorization.annotation.ResourceCrn;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.events.responses.CloudbreakEventV4Response;
+import com.sequenceiq.cloudbreak.auth.crn.CrnResourceDescriptor;
+import com.sequenceiq.cloudbreak.auth.security.internal.TenantAwareParam;
+import com.sequenceiq.cloudbreak.domain.view.StackView;
+import com.sequenceiq.cloudbreak.facade.CloudbreakEventsFacade;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.service.workspace.WorkspaceService;
+import com.sequenceiq.cloudbreak.structuredevent.event.CloudbreakEventService;
+import com.sequenceiq.cloudbreak.structuredevent.event.StructuredEventType;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPOperationDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredEvent;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredNotificationEvent;
+import com.sequenceiq.cloudbreak.validation.ValidCrn;
+import com.sequenceiq.distrox.api.v1.distrox.endpoint.DistroXV1EventEndpoint;
+
+@Controller
+public class DistroXV1EventController implements DistroXV1EventEndpoint {
+
+    @Inject
+    private WorkspaceService workspaceService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private CloudbreakEventsFacade cloudbreakEventsFacade;
+
+    @Override
+    @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.DESCRIBE_DATAHUB)
+    public List<CDPStructuredEvent> getAuditEvents(
+            @ValidCrn(resource = CrnResourceDescriptor.DATAHUB) @ResourceCrn @TenantAwareParam String resourceCrn, Integer page, Integer size) {
+        Long workspaceId = workspaceService.getForCurrentUser().getId();
+        StackView stackView = Optional.ofNullable(stackService.getViewByCrnInWorkspace(resourceCrn, workspaceId)).orElseThrow(notFound("stack", resourceCrn));
```

Comment:
could you init the mdc context? and if necessary could you add logs?

Reply:
Sure I add both of them.